actors/miner/tests/compact_partitions_test.rs

```diff
@@ -0,0 +1,269 @@
+use fil_actor_miner::{
+    expected_reward_for_power, new_deadline_info, pledge_penalty_for_termination,
+    qa_power_for_sector, State, INITIAL_PLEDGE_PROJECTION_PERIOD,
+};
+use fil_actors_runtime::{
+    runtime::{Runtime, RuntimePolicy},
+    test_utils::{expect_abort, expect_abort_contains_message, MockRuntime},
+    EPOCHS_IN_DAY,
+};
+use fvm_ipld_bitfield::BitField;
+use fvm_shared::{clock::ChainEpoch, econ::TokenAmount, error::ExitCode, sector::SectorNumber};
+
+mod util;
+use fvm_shared::bigint::{BigInt, Zero};
+use itertools::Itertools;
+use util::*;
+const PERIOD_OFFSET: ChainEpoch = 100;
+
+fn setup() -> (ActorHarness, MockRuntime) {
+    let big_balance = 20u128.pow(23);
+
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    h.construct_and_verify(&mut rt);
+    rt.balance.replace(TokenAmount::from(big_balance));
+
+    (h, rt)
+}
+
+fn assert_sectors_exists(
+    rt: &MockRuntime,
+    sector_number: SectorNumber,
+    expected_partition: u64,
+    expected_deadline: u64,
+) {
+    let state: State = rt.get_state();
+    assert!(state.get_sector(rt.store(), sector_number).unwrap().is_some());
+
+    let (deadline, pid) = state.find_sector(rt.policy(), rt.store(), sector_number).unwrap();
+    assert_eq!(expected_partition, pid);
+    assert_eq!(expected_deadline, deadline);
+}
+
+fn assert_sectors_not_found(rt: &MockRuntime, sector_number: SectorNumber) {
+    let state: State = rt.get_state();
+    assert!(state.get_sector(rt.store(), sector_number).unwrap().is_none());
+
+    let err = state.find_sector(rt.policy(), rt.store(), sector_number).err().unwrap();
+    assert!(err.to_string().contains("not due at any deadline"));
+}
+
+#[test]
+fn compacting_a_partition_with_both_live_and_dead_sectors_removes_dead_sectors_retains_live_sectors(
+) {
+    let (mut h, mut rt) = setup();
+    rt.set_epoch(200);
+
+    // create 4 sectors in partition 0
+    let sectors_info = h.commit_and_prove_sectors(
+        &mut rt,
+        4,
+        DEFAULT_SECTOR_EXPIRATION,
+        vec![10, 20, 30, 40],
```

Comment:
I think this function now needs to be updated to take a vector of vectors so that it matches the go tests.